src/util.ts

```diff
@@ -1033,7 +1033,22 @@ export class Util {
                         return new VoidType();
                 }
         }
+    }
 
+    /**
+     * Get the extension for the given file path. Basically the part after the final dot, except for
+     * `d.bs` which is treated as single extension
+     */
+    public getExtension(filePath: string) {
+        filePath = filePath.toLowerCase();
+        if (filePath.endsWith('.d.bs')) {
+            return '.d.bs';
+        } else {
+            const parts = filePath.split('.');
```

Comment:
`split` is pretty wasteful.
Maybe get `lastIndexOf('.')` or maybe just test explicitly the extensions you support, like `endsWith('.bs')`, `endsWith('.xml')` and the rest is ''.

Reply:
Sure, I can change that. `lastIndexOf` is much quicker (relatively speaking...neither would have any tangible impact on performance since ops/sec is in the billions for either implementation).
![image](https://user-images.githubusercontent.com/2544493/102613513-9e812000-4100-11eb-9141-6b9942c6e1c7.png)